odp_sdk/tabular_storage_client.py

```diff
@@ -267,14 +236,60 @@ def delete_stage_request(self, resource_dto: ResourceDto, table_stage: TableStag
                 raise OdpResourceNotFoundError("Schema not found") from e
             raise
 
-    def _get_crud_url(self, resource_dto: ResourceDto):
-        if resource_dto.metadata.uuid:
-            return f"{self.tabular_storage_url}/{resource_dto.metadata.uuid}"
+    def select(
+        self,
+        resource_dto: ResourceDto,
+        filter_query: Optional[dict] = None,
+        limit: Optional[int] = None,
+        cursor: Optional[str] = None,
+        paginate: Optional[bool] = None,
+    ) -> Iterable[dict]:
+        """Read data from tabular API
+
+        Args:
+            resource_dto: Dataset manifest
+            filter_query: Filter query in OQS format
+            limit: Limit for the number of rows returned
+            cursor: Pagination token for the next page
+            paginate: Whether or not to paginate the response or use chunked encoding
+
+        Yields:
+            Each row of the data
+        """
+
+        paginate = paginate if paginate is not None else self.paginate_by_default
+
+        if paginate is True:
+            use_limit = self.pagination_size
         else:
-            return f"{self.tabular_storage_url}/catalog.hubocean.io/dataset/{resource_dto.metadata.name}"
+            use_limit = self.chunked_encoding_page_size
+
+        if limit:
+            use_limit = min(use_limit, limit)
```

Comment:
Do I understand correctly that use_limit is hard cap of how many rows we would like to return?
Wondering because of case: if paginate is False, user gives limit higher than chunked_encoding_page_size, we would limit by chunked_encoding_page_size

Reply:
This is correct. We will still paginate somewhat, but with much larger pages. This is to spare be "kind to" the API.